### PR TITLE
no_dresscode variable but this time it doesn't override important slots

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

Makes it so the no_dresscode variable is a thing again, but this time improves it so the loadout doesn't override things like ID, PDA, headset and backpack.
Also makes the proc equip_to_appropriate_slot take a slot blacklist variable.

## Why It's Good For The Game

Because it's annoying having to equip the stuff you want equipped every shift start, and also annoying to see grey outfits laying around everywhere.

## Changelog
:cl:
tweak: enables the no_dresscode variable once again but better this time
/:cl:

### CODE REVIEW

**Comment by atakiya** [comment]
Would also be appreciated if this could be modified to somehow not do the equip noise on the lobby screen.
As of the previous PR, anyone that would join the round with a loadout overwriting the suit, would play the zipper noise in the lobby.

**Comment by Azarak** [comment]
What's the 'initial' variable for here?

**Comment by Seris02** [comment]
Ah, initial is used because in the equip proc used for equipping jobs, it uses the same variable, which prevents the equipping sound from being played.

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -382,9 +383,12 @@
 			ITEM_SLOT_LPOCKET, ITEM_SLOT_RPOCKET,\
 			ITEM_SLOT_DEX_STORAGE\
 		)
-
+	//SKYRAT EDIT CHANGE BEGIN - CUSTOMIZATION
+	if (blacklist)
```
Comment (Azarak) [inline code/modules/mob/mob.dm]:
```suggestion
	//SKYRAT EDIT CHANGE BEGIN - CUSTOMIZATION
	/*
	for(var/slot in slot_priority)
		if(equip_to_slot_if_possible(W, slot, FALSE, TRUE, TRUE, FALSE, FALSE)) //qdel_on_fail = FALSE; disable_warning = TRUE; redraw_mob = TRUE;
	*/
	if (blacklist)
```

Comment (Azarak) [reply code/modules/mob/mob.dm]: Otherwise I wont know what's the original code

**Comment by atakiya** [comment]
I know I'm late with this question but uh, does this still have the option to *not* do it?
From what I understood the previous PR had that option as a preference, and imo its kinda nice to have options.. 👉👈

**Comment by Seris02** [comment]
No it doesn't, as that would really be unreasonable, there's a grey jumpsuit and black shoes in the loadout, which is all this currently overwrites. Or you could just not pick a suit and shoes. Weh.
Also the PR you're speaking of was closed by me because Azarak told me that it was bad and gave me reasons which I agreed with.

**Comment by atakiya** [comment]
> 
> 
> No it doesn't, as that would really be unreasonable, there's a grey jumpsuit and black shoes in the loadout, which is all this currently overwrites. Or you could just not pick a suit and shoes. Weh.

Ah so it does not override job-specific loadouts like engineering jumpsuits or similar - Only assistant?

**Comment by Seris02** [comment]
Yep, only assistant currently, because no_dresscode isn't enabled for the other jobs.

**Review by Gandalf2k15 — APPROVED** [review]
Attempt number 2. All engine are go.

**Pull request merged**